modules/gapi/src/compiler/gcompiler.cpp

```diff
@@ -417,9 +423,10 @@ cv::GCompiled cv::gimpl::GCompiler::produceCompiled(GPtr &&pg)
 
 cv::GStreamingCompiled cv::gimpl::GCompiler::produceStreamingCompiled(GPtr &&pg)
 {
+    std::unique_ptr<GStreamingExecutor> pE(new GStreamingExecutor(std::move(pg), m_args, m_metas));
+
     const auto &outMetas = GModel::ConstGraph(*pg).metadata()
         .get<OutputMeta>().outMeta;
```

Comment:
Your `outMeta` may be missing if you didn't run meta inference. This code will likely throw.

I asked to support _both_ cases here.